website/integrations: improves netbird documentation (#14191)
goauthentik/authentik: website/integrations/services/netbird/index.md

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.
- **Choose a Provider type**: select **OAuth2/OpenID Connect** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Under **Protocol Settings**:
        - Note the **Client ID**, and **slug** values because they will be required later.
        - Set **Client type** to `Public`.

>         - Add two `Strict` redirect URIs: <kbd>http://localhost:53000</kbd> and <kbd>https://<em>netbird.company</em></kbd>.

Reviewer: Can we switch this up to use angle brackets instead?

2. Navigate to **Directory** > **Groups**, and click **`authentik Admins`**.
3. On the top of the group configuration page, switch to the **Users** tab near the top of the page, then click **Add existing user**, and select the service account you just created.

### Create and apply a device token authentication flow

1. Log in to authentik as an admin, and open the authentik Admin interface.

> 2. Navigate to **Flows & Stages** > **Flows** and click **Create**.

Reviewer 2: ```suggestion
2. Navigate to **Flows and Stages** > **Flows** and click **Create**.
```

3. Set the following required configurations:
    - **Name**: provide a name (e.g. `default-device-code-flow`)
    - **Title**: provide a title (e.g. `Device code flow`)
    - **Slug**: provide a slug (e.g `default-device-code-flow`)
    - **Designation**: `Stage Configuration`
    - **Authentication**: `Require authentication`
4. Click **Create**.
5. Navigate to **System** > **Brands** and click the **Edit** icon on the default brand.
6. Set **Default code flow** to the newly created device code flow and click **Update**.

## NetBird configuration

To configure NetBird to use authentik, add the following environment variables to your NetBird deployment:

> ```yaml showLineNumbers

Reviewer: authentik.company is already defined as a placeholder up top. as a result, you don't need to surround it with angle brackets

> ```yaml showLineNumbers

Reviewer: ```suggestion
```yaml title="setup.env"
```

I think

NETBIRD_AUTH_OIDC_CONFIGURATION_ENDPOINT="https://authentik.company/application/o/<application slug>/.well-known/openid-configuration"
NETBIRD_USE_AUTH0=false
NETBIRD_AUTH_CLIENT_ID="<Your Client ID>"

> NETBIRD_AUTH_SUPPORTED_SCOPES="openid profile email offline_access api"

Reviewer 3: I think this variable can be removed. `NETBIRD_AUTH_AUDIENCE` 
This is a public client and there is not Client Secret to set

NETBIRD_IDP_MGMT_EXTRA_PASSWORD="<Your Service Account password>"
```

Restart the NetBird service for the changes to take effect. If using Docker, redeploy the NetBird container for the changes to take effect.

## Configuration verification

To confirm that authentik is properly configured with NetBird, log out and log back in via authentik.

## Troubleshooting

> If using a reverse proxy to access NetBird it's possible to get stuck in a loop where the `/peers` URL will reload. To resolve this set the following environment variables in your NetBird deployment:

Reviewer 3: Maybe add a note here to set it in the setup.env and re-run `./configure.sh` again.

Author: Great suggestion - I've made that change. Thanks again.

NETBIRD_IDP_MGMT_EXTRA_PASSWORD="<Your Service Account password>"
```

Restart the NetBird service for the changes to take effect. If using Docker, redeploy the NetBird container for the changes to take effect.

## Configuration verification

To confirm that authentik is properly configured with NetBird, log out and log back in via authentik.

## Troubleshooting

> If using a reverse proxy to access NetBird it's possible to get stuck in a loop where the `/peers` URL will reload. To resolve this set the following variables in your NetBird `setup.env` file:

Reviewer: The file name is included in the yaml block below. Do we also need to include this here?

Reviewer: ```suggestion
When accessing NetBird through a reverse proxy, you may encounter a loop where the `/peers` URL continuously reloads. To resolve this, set the following variables accordingly:
```
